src/com/owncloud/android/ui/helpers/FileOperationsHelper.java

```diff
@@ -134,6 +136,37 @@ public void openFile(OCFile file) {
     }
 
     /**
+     * Show dialog to allow the user to choose an app to send the private link of an {@link OCFile},
+     * or copy it to clipboard.
+     *
+     * @param context
+     * @param file @param file {@link OCFile} which will be shared with internal users
+     * @param account
+     */
+    public void copyOrSendPrivateLink(Context context, Account account, OCFile file) {
+
+        try {
+
+            // Parse remoteId
+            String remoteId = file.getRemoteId();
```

Comment:
All this block would be better together in a method ```getPermalink()```, or ```getPrivateLink()```; might be in ```OCFile```, and receive the base URL to the server as a parameter. Take into account that the manipulation of  the remote Id is quite specific of private link right now, and the responsibility of this method is not generating the permalink itself, but that of displaying the menu of apps capable of handling the permalink.